mypyc/lib-rt/CPy.h

```diff
@@ -456,6 +456,14 @@ static inline bool CPyFloat_Check(PyObject *o) {
     return PyFloat_Check(o) || PyLong_Check(o);
 }
 
+static inline bool CPy_TypeCheck(PyObject *o, PyObject *type) {
+    return PyObject_TypeCheck(o, (PyTypeObject *)type);
+}
+
+static inline PyObject *CPy_CalculateMetaclass(PyObject *type, PyObject *o) {
+    return (PyObject *)_PyType_CalculateMetaclass((PyTypeObject *)type, o);
```

Comment:
This use of casts is a good way to work around the pointer type mismatches, and it's not a problem if we can't inline this on non-C backends, since this isn't performance critical.